streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -68,36 +69,60 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
   }
 
   def stop(processAllReceivedData: Boolean): Unit = synchronized {
-    if (eventActor == null) return // scheduler has already been stopped
-    logDebug("Stopping JobScheduler")
+    val shutdownExecutor = Executors.newFixedThreadPool(1)
```

Comment:
Its confusing to have another variable "shutdownExecutor" when there is a "jobExecutor". It might be a good idea to rename this to something like "shutdownThreadPool`.